TestAssemblies/AssemblyToProcess/ClassWithOnChangedCalculatedProperty.cs

```diff
@@ -0,0 +1,17 @@
+using System.ComponentModel;
+
+public class ClassWithOnChangedCalculatedProperty :
+    INotifyPropertyChanged
+{
+    public bool OnProperty2ChangedCalled;
+
+    public string Property1 { get; set; }
+    public string Property2 => $"{Property1}!";
```

Comment:
I'd add a `public string Property3 => $"{Property2}!";` with a `void OnProperty3Changed()` and a `OnProperty3ChangedCalled` and the associated assert to test second-order dependencies.

Reply:
Done